core/contracts_stateful_test.go

```diff
@@ -0,0 +1,448 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package core
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ava-labs/coreth/core/rawdb"
+	"github.com/ava-labs/coreth/core/state"
+	"github.com/ava-labs/coreth/core/vm"
+	"github.com/ava-labs/coreth/params"
+	"github.com/ava-labs/coreth/vmerrs"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/holiman/uint256"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	big0                        = big.NewInt(0)
+	genesisContractAddr         = common.HexToAddress("0x0100000000000000000000000000000000000000")
+	NativeAssetBalanceAddr      = vm.NativeAssetBalanceAddr
+	PackNativeAssetBalanceInput = vm.PackNativeAssetBalanceInput
+	NativeAssetCallAddr         = vm.NativeAssetCallAddr
+	PackNativeAssetCallInput    = vm.PackNativeAssetCallInput
+)
+
+type StateDB = vm.StateDB
+
+func TestStatefulPrecompile(t *testing.T) {
```

Comment:
this test has to move out of core/vm, since we want to get the statedb with the hook configured.